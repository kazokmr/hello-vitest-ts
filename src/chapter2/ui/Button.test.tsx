import { describe, expect, test } from "vitest";
import * as renderer from "react-test-renderer";
import { Button } from "./Button";
import { render } from "@testing-library/react";
import userEvent from "@testing-library/user-event";

describe("Button", () => {
  test("renders correctly with react-test-renderer", () => {
    const button = renderer.create(<Button />);
    expect(button.toJSON()).toMatchSnapshot();
  });

  test("changes the button text upon clicking the button using React Testing Library", async () => {
    const button = render(<Button />);
    const user = userEvent.setup();
    await user.click(button.getByRole("button", { name: "ON" }));
    expect(button.getByRole("button", { name: "OFF" })).toBeTruthy();
    await user.click(button.getByRole("button", { name: "OFF" }));
    expect(button.getByRole("button", { name: "ON" })).toBeTruthy();
  });
});
