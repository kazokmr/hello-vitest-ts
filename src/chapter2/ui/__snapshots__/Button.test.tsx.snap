// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Button > renders correctly with react-test-renderer 1`] = `
<button
  className="primary"
  onClick={[Function]}
>
  ON
</button>
`;
